deps: update golang/x/net to pickup hpack fix.

mberhault opened the pull request:
This points to vendored deps with HEAD golang.org/x/net/
We need to pickup
https://github.com/golang/net/commit/05d3205156cb1bdc096578d6457fa161cd950aa2
to fix https://github.com/cockroachdb/cockroach/issues/17052

Commit:
deps: update golang/x/net to pickup hpack fix.

This points to vendored deps with HEAD golang.org/x/net/
We need to pickup
https://github.com/golang/net/commit/05d3205156cb1bdc096578d6457fa161cd950aa2
to fix https://github.com/cockroachdb/cockroach/issues/17052

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/17080)
<!-- Reviewable:end -->

mberhault commented:
depends on https://github.com/cockroachdb/vendored/pull/4

bdarnell reviewed (approved):
I don't think we do PRs on the vendored repo, we just push whatever is needed for the submodule to work.

mberhault commented:
oh, uh.. too late :)

mberhault commented:
and for info, I put a build with this change on `cockroach-omega-0001`, left all other nodes at the alpha candidate sha, enabled tracing, and restarted all nodes (full shutdown-restart).

Graph of go allocated memory:
![omega_mem_diff](https://user-images.githubusercontent.com/5901480/28328619-23880b6c-6bb5-11e7-841e-c9f7eb045f60.png)

pprof of nodes 1 (with fix) and 2 (without fix): (cumulative)
```
$ go tool pprof -cum -text cockroach.1 1.memprof.2017-07-18T16_28_52.626 |head
174.22MB of 200.28MB total (86.99%)
Dropped 1350 nodes (cum <= 1MB)
      flat  flat%   sum%        cum   cum%
         0     0%     0%   196.14MB 97.93%  runtime.goexit
         0     0%     0%    84.53MB 42.20%  github.com/cockroachdb/cockroach/pkg/storage.(*endCmds).done
   69.01MB 34.46% 34.46%    77.03MB 38.46%  github.com/cockroachdb/cockroach/pkg/storage.(*timestampCache).AddRequest
         0     0% 34.46%    70.21MB 35.05%  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleStream
         0     0% 34.46%    70.21MB 35.05%  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).processUnaryRPC
         0     0% 34.46%    70.21MB 35.05%  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1
         0     0% 34.46%    69.67MB 34.79%  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1

$ go tool pprof -cum -text cockroach.2 2.memprof.2017-07-18T16_29_15.149|head
795.82MB of 892.81MB total (89.14%)
Dropped 1332 nodes (cum <= 4.46MB)
      flat  flat%   sum%        cum   cum%
         0     0%     0%   882.81MB 98.88%  runtime.goexit
   45.51MB  5.10%  5.10%   429.14MB 48.07%  runtime.mapassign
         0     0%  5.10%   408.60MB 45.77%  github.com/cockroachdb/cockroach/vendor/golang.org/x/net/http2/hpack.(*dynamicTable).add
  379.12MB 42.46% 47.56%   379.12MB 42.46%  runtime.hashGrow
         0     0% 47.56%   335.11MB 37.53%  github.com/cockroachdb/cockroach/pkg/kv.(*grpcTransport).SendNext.func1
         0     0% 47.56%   335.11MB 37.53%  github.com/cockroachdb/cockroach/pkg/kv.(*grpcTransport).SendNext.func1.1
         0     0% 47.56%   325.69MB 36.48%  github.com/cockroachdb/cockroach/vendor/github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingClientInterceptor.func1
```
